js/hw_3_js_react_1/src/components/Board/Board.jsx

```diff
@@ -0,0 +1,192 @@
+import React, { useState, useEffect } from "react";
+
+import { generateNewFood } from "../../function/Food/generateNewFood";
+import { snakeGrow } from "../../function/Snake/snakeGrow";
+import { snakeBodyTail } from "../../function/Snake/snakeBodyTail";
+import {
+  getIsSnakeHeadOutsideBoard,
+  getIsSnakeHeadOnBody,
+  getIsTimeOut,
+} from "../../function/GameOver/gameOver";
+import { GetEat } from "../../function/Snake/GetEat";
+import { GameTimer } from "../GameTimer/GameTimer";
+import { GameScore } from "../GameScore/GameScore"
+
+import "./Board.css";
+
+const boardWidth = 25;
+const board = Array(boardWidth).fill(Array(boardWidth).fill(0));
+const initialSnakeBody = [
+  [10, 10],
+  [10, 11],
+  [10, 12],
+  [10, 13],
+  [10, 14],
+];
+const initialFood = [[]];
+const speed = 200;
+const initialTime = 120
+
+const Board = () => {
+  const [snake, setSnake] = useState([initialSnakeBody]);
+  const [food, setFood] = useState(initialFood);
+  const [direction, setDirection] = useState("");
+  const [gameOver, setGameOver] = useState(false);
+  const [score, setScore] = useState(0);
+  const [time, setTime] = useState(initialTime);
+  const [start, setStart] = useState(false);
+  const [pause, setPause] = useState(false);
+
+  useEffect(() => {
+    if (gameOver) {
+      return;
+    }
+    document.addEventListener("keydown", changeDirection);
+    return () => document.removeEventListener("keydown", changeDirection);
+  });
+
+  useEffect(() => {
+    if (gameOver) {
+      setStart(false);
+    }
+    if (start && !pause) {
+      const intervalId = setInterval(() => {
+        setTime((prevTime) => prevTime - 1);
+      }, 1000);
+      return () => clearInterval(intervalId);
+    }
+  }, [start, time, gameOver, pause]);
+
+  useEffect(() => {
+    if (gameOver) {
+      return;
+    }
+    if (start && !gameOver && !pause) {
+      const intervalId = setTimeout(() => {
+        snakeMove();
+      }, speed)
+
+      return () => {
+        clearInterval(intervalId);
+      };
+    }
+  }, [direction, snake, gameOver, pause, start]);
+
+  function play() {
+    setSnake(initialSnakeBody);
+    setFood(generateNewFood(snake));
+    setDirection("ArrowUp");
+    setScore(0);
+    setGameOver(false);
+    setTime(initialTime);
+    setStart(true);
+  }
+
+  function changeDirection(e) {
+    if (e.key === "ArrowLeft" && direction !== "ArrowRight") {
+      setDirection("ArrowLeft");
+    } else if (e.key === "ArrowUp" && direction !== "ArrowDown") {
+      setDirection("ArrowUp");
+    } else if (e.key === "ArrowRight" && direction !== "ArrowLeft") {
+      setDirection("ArrowRight");
+    } else if (e.key === "ArrowDown" && direction !== "ArrowUp") {
+      setDirection("ArrowDown");
+    }
+    if (e.code === "Space") {
+      setPause(!pause);
+    }
+  }
+
+  function snakeMove() {
+      const snakeBodyBefore = [...snake];
```

Comment:
This function is quite big - is it possible to move it out of the component? Think about it.